packages/flutter/test/widgets/scroll_aware_image_provider_test.dart

```diff
@@ -0,0 +1,335 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:flutter/widgets.dart';
+
+import '../painting/image_test_utils.dart';
+import '../painting/mocks_for_image_cache.dart' show TestImage;
+
+void main() {
+  tearDown(() {
+    imageCache.clear();
+  });
+
+  RecordingPhysics _findPhysics(WidgetTester tester) {
+    return Scrollable.of(find.byType(TestWidget).evaluate().first).position.physics as RecordingPhysics;
+  }
+
+  ScrollMetrics _findMetrics(WidgetTester tester) {
+    return Scrollable.of(find.byType(TestWidget).evaluate().first).position;
+  }
+
+  testWidgets('DisposableBuildContext asserts on disposed state', (WidgetTester tester) async {
+    final GlobalKey<TestWidgetState> key = GlobalKey<TestWidgetState>();
+    await tester.pumpWidget(TestWidget(key));
+
+    final TestWidgetState state = key.currentState;
+    expect(state.mounted, true);
+
+    final DisposableBuildContext context = DisposableBuildContext(state);
+    expect(context.debugValidate(), true);
```

Comment:
Instead of marking debugValidate visible for tests, could we just test the same by actually accessing the context property and checking that we do or don't get an assert?

(That would also prevent us from ever removing that assert from the context getter)